src/transaction.ts

```diff
@@ -838,6 +1020,44 @@ class Transaction extends DatastoreRequest {
 
     this.save(entities);
   }
+
+  /**
+   * If the transaction has not begun yet then this function ensures the transaction
+   * has started before running the function provided as a parameter.
```

Comment:
I'd probably re-word this, something like: 

```
Some rpc require that the transaction has been started (i.e, has a valid id) before they can be sent. #withBeginTransaction acts as a wrapper over those functions. 

If the function has not begun yet, `#withBeginTransaction` will first send an rpc to begin the transaction, and then execute the wrapped function. If it has begun, the wrapped function will be called directly instead

If an error is encountered during the transaciton begin call, the callback will be executed instead of the wrapped function
```